Auteur: Jean-Claude Moissinac, juin 2021

Note 1: les valeurs mentionnées dans les blocs de commentaires sont celles obtenues au 15/6/2021; elles peuvent évoluer au fil de mises à jour effectuées dans Wikidata

Note 2: pour des considérations techniques sur la réalisation de ce notebook, voir le blog http://www.bobdc.com/blog/jupytersparql/

Dans ce notebook, je vais utiliser le point d'accès sparql de Wikidata pour explorer les mouvements artistiques connus de Wikidata et en rapport avec les créations répertoriées dans la base Joconde et décrites dans Wikidata. 
Pour cela, je commence par définir le point d'accès (endpoint), le format d'échange et le format d'affichage.

In [10]:
%endpoint http://query.wikidata.org/bigdata/namespace/wdq/sparql
%format any
%display table
# The "magics" don't have to be in a separate cell from the actual query. 

Endpoint set to: http://query.wikidata.org/bigdata/namespace/wdq/sparql
Request format: ANY
Display: table

Pour se limiter à des créations présentes dans la base Joconde, la première idée est de trouver toutes les entités qui ont un lien avec la base Joconde (wd:Q809825). 

In [15]:
select distinct ?item
where
{
?item ?p wd:Q809825 .
}

item http://www.wikidata.org/reference/00c92e8387b33e628ddf9ce73d11387f83e8c39c http://www.wikidata.org/reference/00d1ebd194d50105b72565969e93f75190ff0d15 http://www.wikidata.org/reference/00dce694e9db108e5b1df9cd65edc8c2301063ef http://www.wikidata.org/reference/00eeab55f94623bb23ad1b377c27f0a46d51dcdd http://www.wikidata.org/reference/00ffbe2f497182ae8de966cb34ec8e94ececaa6a http://www.wikidata.org/reference/0105ae6a4cd4dadfe46cace9f9e7eb0fb4f8bc70 http://www.wikidata.org/reference/0117ef0bbb282d2cd4bb5d6c5227b048902f7c53 http://www.wikidata.org/reference/0126165f7f78c3d798898ef0ffea2762008e8068 http://www.wikidata.org/reference/012a07adf637bdd7ed36b307a863923fc720f4cd http://www.wikidata.org/reference/01387808ad221336ebd5087b8941177030ee322a http://www.wikidata.org/reference/01551b2e2de953040a042504cfaad53fc606cb32 http://www.wikidata.org/reference/015629bef9837b2b2fff0c2854a683e885d7f3f8 http://www.wikidata.org/reference/01645261e0d8c6fbe1c7f5d676a43365ecdda982 http://www.wikidata.org/reference/017f87d500f8bc41e30dc305f6e26edafe98f3fa http://www.wikidata.org/reference/0184ed35150644b857f36802067f9f0407e9968b http://www.wikidata.org/reference/01a0edbdccc18805118a223bb57e3a4da55225ee http://www.wikidata.org/reference/01a8f7af26dda95519426f6f8555f24ddf3baa86 http://www.wikidata.org/reference/01aa951cab52d954a7da21adc6e01285f22685e1 http://www.wikidata.org/reference/01ab3275c44cbe51423a37d508d3cb09dba4ca0c http://www.wikidata.org/reference/01af252d040711a8c6f9467edbe9c25870ccb706 Total: 4511, Shown: 20

Cela donne 4511 entités (au 15/6/2021). Certaines sont des propriétés; d'autres des créateurs. Mais, la plupart sont des références à une source de données; ces références servent à Wikidata à indiquer d'où viennent les faits décrits pat les triplets concernés par cette référence. Par exemple:

In [61]:
%format ANY
%display table

Request format: ANY
Display: table

In [63]:
select ?link
where
{
<http://www.wikidata.org/reference/ec603861711ed65b7acc54284085fb2a2e686028> pr:P854 ?link
}

link http://www.culture.gouv.fr/public/mistral/joconde_fr?ACTION=CHERCHER&FIELD_1=REF&VALUE_1=000PE002808 Total: 1, Shown: 1

me donne par la propriété pr:P854 un lien vers la base Joconde qui correspond à une œuvre précise.
Ce qui suggère de chercher les entités qui ont un lien avec ces références.

In [59]:
select distinct ?item
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?item ?pref ?ref
}

item http://www.wikidata.org/entity/statement/Q29655733-2313515E-6060-4CA5-BBF6-09BBCF721716 http://www.wikidata.org/entity/statement/Q29655733-23F3B678-C192-4DA4-BD89-5FE71CCEEF1F http://www.wikidata.org/entity/statement/Q29655733-4267C92F-59CE-49DC-A9CF-EA40CC616D43 http://www.wikidata.org/entity/statement/Q29655733-6657AF1D-71BB-4D6B-8246-DF8CB2F98103 http://www.wikidata.org/entity/statement/Q29655733-8CE0143E-AA32-4EDB-91A4-4290502B7F8B http://www.wikidata.org/entity/statement/Q29655733-D632577E-57E7-4FA2-8A2D-B33B54770F02 http://www.wikidata.org/entity/statement/Q29655733-DDF30572-E638-456E-BA3A-59718313E925 http://www.wikidata.org/entity/statement/Q29655733-EACFAA55-E8BF-4EBC-9F7F-78669D25E178 http://www.wikidata.org/entity/statement/Q29848244-121144D6-DA25-4E80-8FBB-3AD2652CB21B http://www.wikidata.org/entity/statement/Q29848244-24A539A2-82F8-4857-B40A-AA7A155D090B http://www.wikidata.org/entity/statement/Q29848244-7E39E484-F895-468A-ABEB-F58F28288182 http://www.wikidata.org/entity/statement/Q29848244-9332D0FB-5D29-49CA-A693-1CA76A7A4798 http://www.wikidata.org/entity/statement/Q29848244-C94F39D7-6A4D-4725-BE9F-52DB58554D85 http://www.wikidata.org/entity/statement/Q29848244-CE0963FC-4E5C-425E-85EC-05D998235259 http://www.wikidata.org/entity/statement/Q29848244-DD00CC2C-7608-49EF-8593-8B60D4904022 http://www.wikidata.org/entity/statement/Q29848244-EFC35C5A-3130-4D74-8572-4FF58CBED945 http://www.wikidata.org/entity/statement/Q29647966-30F9710C-453B-485C-A709-E814112A7801 http://www.wikidata.org/entity/statement/Q29647966-344D94EA-746F-459F-955B-85270F746B81 http://www.wikidata.org/entity/statement/Q29647966-7D922312-1D99-4228-A90A-F453DA0025EC http://www.wikidata.org/entity/statement/Q29647966-A37391B9-6F43-42F4-B408-9AB13BEDAEDD Total: 80216, Shown: 20

donne 80216 entités (au 15/6/2021), qui sont tous des 'statements', tels que définis par Wikidata. Les statements sont des 'faits' relatifs à une entité. Voyons sur un exemple quel genre d'entité est concerné:

In [18]:
select ?item ?p
where
{
?item ?p <http://www.wikidata.org/entity/statement/Q29656881-4BBBE7E8-1606-4332-84FA-61C2DBD2F32A>
}

item 
 p http://www.wikidata.org/entity/Q29656881 
 http://www.wikidata.org/prop/P195 Total: 1, Shown: 1

Relié au statement par la propriété 'collection' (p:P195), je trouve une peinture: wd:Q29656881.
Je trouve 9169 entités (au 15/6/2021)-essentiellement des oeuvres ou créations- reliées à ces statements:

In [19]:
select distinct ?item
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
}

item http://www.wikidata.org/entity/Q2892964 http://www.wikidata.org/entity/Q18572937 http://www.wikidata.org/entity/Q18573608 http://www.wikidata.org/entity/Q28950338 http://www.wikidata.org/entity/Q29644503 http://www.wikidata.org/entity/Q29644739 http://www.wikidata.org/entity/Q29645278 http://www.wikidata.org/entity/Q29651692 http://www.wikidata.org/entity/Q29656165 http://www.wikidata.org/entity/Q29656881 http://www.wikidata.org/entity/Q29842277 http://www.wikidata.org/entity/Q29842341 http://www.wikidata.org/entity/Q29855894 http://www.wikidata.org/entity/Q29864702 http://www.wikidata.org/entity/Q29643931 http://www.wikidata.org/entity/Q29644923 http://www.wikidata.org/entity/Q29650151 http://www.wikidata.org/entity/Q29654498 http://www.wikidata.org/entity/Q29656286 http://www.wikidata.org/entity/Q29656399 Total: 9169, Shown: 20

Je cherche s'il y a des mouvements directement associés aux œuvres rattachées à Joconde ou aux créateurs associés à ces œuvres.

Pour les entités directement liées à ces statements:

In [27]:
select ?mvt
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
?item wdt:P135 ?mvt .
}

mvt http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q122960 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q1474884 http://www.wikidata.org/entity/Q200034 http://www.wikidata.org/entity/Q200034 Total: 2702, Shown: 20

permet de trouver 2701 entités (au 15/6/2021) reliées à un mouvement. Et:

In [36]:
select  ?mvt ?mvtLabel (count(?item) as ?c)
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
?item wdt:P135 ?mvt .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". } 
}
group by ?mvt ?mvtLabel
order by desc(?c)

mvt 
 mvtLabel 
 c http://www.wikidata.org/entity/Q14378 
 neoclassicism 
 439 http://www.wikidata.org/entity/Q189458 
 academic art 
 209 http://www.wikidata.org/entity/Q170292 
 classicism 
 201 http://www.wikidata.org/entity/Q443153 
 Flemish Primitives 
 197 http://www.wikidata.org/entity/Q122960 
 Rococo 
 188 http://www.wikidata.org/entity/Q37853 
 Baroque 
 146 http://www.wikidata.org/entity/Q131808 
 Mannerism 
 139 http://www.wikidata.org/entity/Q37068 
 Romanticism 
 120 http://www.wikidata.org/entity/Q429811 
 early Christian art and architecture 
 114 http://www.wikidata.org/entity/Q1404472 
 Italian Renaissance 
 112 http://www.wikidata.org/entity/Q1362251 
 Roman sculpture 
 93 http://www.wikidata.org/entity/Q42865 
 Orientalism 
 76 http://www.wikidata.org/entity/Q2352880 
 Dutch Golden Age painting 
 72 http://www.wikidata.org/entity/Q1474884 
 High Renaissance 
 69 http://www.wikidata.org/entity/Q40415 
 Impressionism 
 52 http://www.wikidata.org/entity/Q947129 
 Gothic painting 
 42 http://www.wikidata.org/entity/Q1747618 
 ancient Greek art 
 36 http://www.wikidata.org/entity/Q610687 
 Venetian school 
 36 http://www.wikidata.org/entity/Q1351624 
 Flamboyant 
 35 http://www.wikidata.org/entity/Q164800 
 Symbolism 
 34 Total: 54, Shown: 20

me montre que 54 mouvements sont référencés et le nombre d'oeuvres liées à chaque mouvement.

Pour les créateurs, je trouve 3725 créateurs:

In [31]:
select distinct ?creator ?creatorLabel
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
?item wdt:P170 ?creator
SERVICE wikibase:label { bd:serviceParam wikibase:language "en". } 
}

creator 
 creatorLabel http://www.wikidata.org/entity/Q130039 
 Cornelis Saftleven http://www.wikidata.org/entity/Q271676 
 Charles Le Brun http://www.wikidata.org/entity/Q2113338 
 Prosper Lafaye http://www.wikidata.org/entity/Q315996 
 Jan van Goyen http://www.wikidata.org/entity/Q1341719 
 François-Léon Benouville t2304213301 
 t2304213301 http://www.wikidata.org/entity/Q163543 
 François Gérard http://www.wikidata.org/entity/Q334946 
 Pierre-Henri de Valenciennes http://www.wikidata.org/entity/Q707729 
 Jean François de Troy http://www.wikidata.org/entity/Q975135 
 Noël Coypel http://www.wikidata.org/entity/Q23380 
 Jean-Auguste-Dominique Ingres http://www.wikidata.org/entity/Q5432 
 Francisco Goya http://www.wikidata.org/entity/Q150679 
 Anthony van Dyck http://www.wikidata.org/entity/Q180932 
 François Boucher http://www.wikidata.org/entity/Q5163689 
 Constantijn Netscher t2303411425 
 t2303411425 t2303995166 
 t2303995166 t2304061531 
 t2304061531 http://www.wikidata.org/entity/Q213612 
 Jacob van Ruisdael http://www.wikidata.org/entity/Q944962 
 Achille Etna Michallon Total: 3725, Shown: 20

reliés à 99 mouvements:

In [38]:
select distinct ?mvt ?mvtLabel (count(?item) as ?c)
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
?item wdt:P170 ?creator .
?creator wdt:P135 ?mvt
         SERVICE wikibase:label { bd:serviceParam wikibase:language "en". } 
}
group by ?mvt ?mvtLabel
order by desc(?c)

mvt 
 mvtLabel 
 c http://www.wikidata.org/entity/Q37853 
 Baroque 
 5346 http://www.wikidata.org/entity/Q37068 
 Romanticism 
 3667 http://www.wikidata.org/entity/Q164800 
 Symbolism 
 3208 http://www.wikidata.org/entity/Q14378 
 neoclassicism 
 3196 http://www.wikidata.org/entity/Q122960 
 Rococo 
 2043 http://www.wikidata.org/entity/Q170292 
 classicism 
 1754 http://www.wikidata.org/entity/Q1994273 
 Flemish Baroque painting 
 1501 http://www.wikidata.org/entity/Q2642826 
 Realism 
 1424 http://www.wikidata.org/entity/Q10857409 
 Realism 
 1357 http://www.wikidata.org/entity/Q808561 
 Baroque painting 
 1342 http://www.wikidata.org/entity/Q40415 
 Impressionism 
 1173 http://www.wikidata.org/entity/Q143357 
 Barbizon school 
 985 http://www.wikidata.org/entity/Q2352880 
 Dutch Golden Age painting 
 888 http://www.wikidata.org/entity/Q2614483 
 Pre-romanticism 
 875 http://www.wikidata.org/entity/Q189458 
 academic art 
 812 http://www.wikidata.org/entity/Q42865 
 Orientalism 
 811 http://www.wikidata.org/entity/Q131808 
 Mannerism 
 659 http://www.wikidata.org/entity/Q443153 
 Flemish Primitives 
 341 http://www.wikidata.org/entity/Q200034 
 pointillism 
 336 http://www.wikidata.org/entity/Q150091 
 neo-impressionism 
 324 Total: 99, Shown: 20

Enfin je trouve un maximum de 6 mouvements pour un créateur, Emile Bernard (wd:Q264193). En fait, 627 créateurs sont associés à au moins un mouvement, généralement un seul.

In [39]:
select ?creator  ?creatorLabel (count(distinct ?mvt) as ?c)
where
{
?ref ?p wd:Q809825 .
filter(contains(str(?ref), "http://www.wikidata.org/reference/"))
?statement ?pref ?ref .
?item ?ps ?statement .
?item wdt:P170 ?creator .
?creator wdt:P135 ?mvt
SERVICE wikibase:label { bd:serviceParam wikibase:language "en". } 
}
group by ?creator ?creatorLabel
order by desc(?c)

creator 
 creatorLabel 
 c http://www.wikidata.org/entity/Q264193 
 Émile Bernard 
 6 http://www.wikidata.org/entity/Q37693 
 Paul Gauguin 
 5 http://www.wikidata.org/entity/Q5589 
 Henri Matisse 
 5 http://www.wikidata.org/entity/Q1063574 
 Charles Angrand 
 4 http://www.wikidata.org/entity/Q212499 
 Jean-Léon Gérôme 
 4 http://www.wikidata.org/entity/Q9440 
 Paolo Veronese 
 4 http://www.wikidata.org/entity/Q351746 
 Joaquín Sorolla 
 4 http://www.wikidata.org/entity/Q326606 
 Paul Sérusier 
 4 http://www.wikidata.org/entity/Q34013 
 Georges Seurat 
 3 http://www.wikidata.org/entity/Q310715 
 Henri Fantin-Latour 
 3 http://www.wikidata.org/entity/Q5593 
 Pablo Picasso 
 3 http://www.wikidata.org/entity/Q177978 
 Henri-Jean Guillaume Martin 
 3 http://www.wikidata.org/entity/Q33978 
 Robert Delaunay 
 3 http://www.wikidata.org/entity/Q8459 
 Giorgione 
 3 http://www.wikidata.org/entity/Q630973 
 Albert Lebourg 
 3 http://www.wikidata.org/entity/Q44007 
 Paul Klee 
 3 http://www.wikidata.org/entity/Q122382 
 Henry Fuseli 
 3 http://www.wikidata.org/entity/Q23380 
 Jean-Auguste-Dominique Ingres 
 2 http://www.wikidata.org/entity/Q271676 
 Charles Le Brun 
 2 http://www.wikidata.org/entity/Q148475 
 Jean-Baptiste Camille Corot 
 2 Total: 627, Shown: 20